pkg/util/azureclient/resources.go

```diff
@@ -11,6 +11,7 @@ import (
 type DeploymentsClient interface {
 	CreateOrUpdate(ctx context.Context, resourceGroupName string, deploymentName string, parameters resources.Deployment) (result resources.DeploymentsCreateOrUpdateFuture, err error)
 	Client
+	DeploymentsClientAddons
```

Comment:
CreateOrUpdate() and Client should be removed from the interface (will imply a light refactor elsewhere in our codebase)